### PR TITLE
8305758: Update the JAR tool man page to indicate -i/--generate-file is deprecated 

### PR DESCRIPTION
**Pull request by LanceAndersen**
Hi all,

Please review this MR which updates the jar man page to to indicate that  --generate-index/-I are deprecated and may be removed in a future release

Support for JAR Index is being removed via JDK-8302819.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305758](https://bugs.openjdk.org/browse/JDK-8305758): Update the JAR tool man page to indicate -i/--generate-file is deprecated

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13400/head:pull/13400` \
`$ git checkout pull/13400`

Update a local copy of the PR: \
`$ git checkout pull/13400` \
`$ git pull https://git.openjdk.org/jdk.git pull/13400/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13400`

View PR using the GUI difftool: \
`$ git pr show -t 13400`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13400.diff">https://git.openjdk.org/jdk/pull/13400.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13400#issuecomment-1500863466)

### CODE REVIEW

**Comment by eirbjo** [comment]
The JBS issue name should be updated to use a lowercase `-i` and this PR's title should be updated to match the JBS title.

**Comment by eirbjo** [comment]
Is the `compiler` label appropriate for this PR?

**Comment by LanceAndersen** [comment]
> The JBS issue name should be updated to use a lowercase `-i` and this PR's title should be updated to match the JBS title.

Lack of coffee and doing this before 7am this morning  is my only excuse

**Comment by LanceAndersen** [comment]
> Is the `compiler` label appropriate for this PR?

This was done by the bots,  not sure why but not really an issue

File: src/jdk.jartool/share/man/jar.1 [hunk 1]
```diff
@@ -35,7 +35,7 @@
 . ftr VB CB
 . ftr VBI CBI
 .\}
-.TH "JAR" "1" "2023" "JDK 21-ea" "JDK Commands"
+.TH "JAR" "1" "2023" "JDK 21-internal" "JDK Commands"
```
Comment (AlanBateman) [inline src/jdk.jartool/share/man/jar.1:38]: I assume you didn't mean to edit this.

Comment (LanceAndersen) [reply src/jdk.jartool/share/man/jar.1]: > I assume you didn't mean to edit this.

Updated, thank you.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**